### PR TITLE
Update dependencies and remove rx from security module

### PR DESCRIPTION
**Pull request by rmeissner**
- Migrate security module from RxJava to Kotlin Coroutines
- Bump several libs to newest version
- Remove Fingerprint support in security module
- Add `removePassword()` to EncryptionManager to start over if checksum or password in prefs are out of sync

### CODE REVIEW

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 1]
```diff
@@ -82,26 +76,24 @@ class AesEncryptionManager(
...
-                key = previousKey?.let {
-                    decryptAppKey(it)
-                } ?: generateKey()
-                key?.let {
-                    preferencesManager.prefs.edit { putString(PREF_KEY_PASSWORD_ENCRYPTED_APP_KEY, encrypt(passwordKey, it).toString()) }
-                    preferencesManager.prefs.edit { putString(PREF_KEY_PASSWORD_CHECKSUM, generateCryptedChecksum(passwordKey)) }
-                }
-
-                key != null
+    override fun setupPassword(newPassword: ByteArray, oldPassword: ByteArray?): Boolean {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:79]: A bit concerned that this is no longer called in a different thread. Couldn't we mark it as `suspend` to enforce this being from a coroutine context?

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 2]
```diff
@@ -110,24 +102,20 @@ class AesEncryptionManager(
         return keyStorage.store(generatedPassword)
     }
 
-    override fun unlocked(): Single<Boolean> {
-        return Single.fromCallable {
-            synchronized(keyLock) {
-                key != null
-            }
+    override fun unlocked(): Boolean {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:105]: same as above ☝️

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 3]
```diff
@@ -110,24 +102,20 @@ class AesEncryptionManager(
...
-        return Single.fromCallable {
-            synchronized(keyLock) {
-                // If we have no password set (no checksum stored, we cannot unlockWithPassword
-                val checksum = preferencesManager.prefs.getString(PREF_KEY_PASSWORD_CHECKSUM, null) ?: return@fromCallable false
-                val passwordKey = buildPasswordKeyIfValid(password, checksum) ?: return@fromCallable false
-                key = decryptAppKey(passwordKey) ?: return@fromCallable false
-                key != null
-            }
-        }.subscribeOn(Schedulers.io())
+    override fun unlockWithPassword(password: ByteArray): Boolean {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:111]: Same as above ☝️

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 4]
```diff
@@ -208,38 +195,33 @@ class AesEncryptionManager(
...
-                        }
-                        if (key != null) FingerprintUnlockSuccessful() else throw FingerprintUnlockError()
-                    }
-                    is AuthenticationFailed -> FingerprintUnlockFailed()
-                    is AuthenticationHelp -> FingerprintUnlockHelp(authResult.helpString)
+    override suspend fun unlockWithFingerprint(): FingerprintUnlockResult {
+        val cryptedData = CryptoData.fromString(
+            preferencesManager.prefs.getString(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY, null) ?: throw FingerprintUnlockError()
+        )
+        val authResult = fingerprintHelper.authenticate(cryptedData.iv)
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:202]: In case you didn't know, you can put this whole line within the parenthesis of the `when()` :D

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 5]
```diff
@@ -208,38 +195,33 @@ class AesEncryptionManager(
...
+            is AuthenticationFailed -> FingerprintUnlockFailed()
+            is AuthenticationHelp -> FingerprintUnlockHelp(authResult.helpString)
+        }
+    }
 
-    override fun isFingerPrintSet(): Single<Boolean> =
-        fingerprintHelper.isKeySet()
-            .map { it && preferencesManager.prefs.getString(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY, null) != null }
-            .onErrorReturn { false }
+    override fun isFingerPrintSet(): Boolean =
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:215]: Should this somehow be `suspend` still?

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 6]
```diff
@@ -208,38 +195,33 @@ class AesEncryptionManager(
...
+        nullOnThrow { fingerprintHelper.isKeySet() } == true &&
+                preferencesManager.prefs.getString(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY, null) != null
 
-    override fun clearFingerprintData(): Completable =
-        Completable.fromCallable {
-            preferencesManager.prefs.edit {
-                remove(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY)
-            }
-        }.andThen(fingerprintHelper.removeKey())
+    override fun clearFingerprintData() {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:219]: Same here. Maybe making this `suspend` would be nice.

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 7]
```diff
@@ -208,38 +195,33 @@ class AesEncryptionManager(
...
+        preferencesManager.prefs.edit {
+            remove(PREF_KEY_FINGERPRINT_ENCRYPTED_APP_KEY)
+        }
+        fingerprintHelper.removeKey()
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:223]: My understanding is that `Completable.andThen` only executes on success. If for some reason `remove()` fails ...we still throw and everything behaves the same, right?

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt [hunk 8]
```diff
@@ -35,14 +37,15 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
         return keyGenerator.generateKey()
     }
 
-    override fun removeKey(): Completable = Completable.fromCallable {
+    override fun removeKey() {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:40]: Missing suspend. Sorry for all the duplicate comments.

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt [hunk 9]
```diff
@@ -51,10 +54,10 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
         return keyStore.getKey(FINGERPRINT_KEY, null) as? SecretKey ?: createKey()
     }
 
-    override fun isKeySet(): Single<Boolean> = Single.fromCallable {
+    override fun isKeySet(): Boolean {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:57]: `suspend` ? Sorry again, feel free to just resolve all these sort of comments. I am leaving these as sort of `bookmarks` for you.

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt [hunk 10]
```diff
@@ -68,53 +71,35 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
...
-                this.emitter = emitter
-                val signal = CancellationSignal()
-                fingerprintManager.authenticate(cryptoObject, 0, signal, this, null)
-                emitter.setCancellable {
-                    if (!signal.isCanceled) {
-                        signal.cancel()
-                        this.emitter = null
+    @SuppressLint("MissingPermission")
+    @RequiresApi(Build.VERSION_CODES.M)
+    override suspend fun authenticate(iv: ByteArray?): AuthenticationResult {
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:76]: what does `iv` stand for?

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt [hunk 11]
```diff
@@ -68,53 +71,35 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
...
+        val cryptoObject = FingerprintManagerCompat.CryptoObject(createCipher(iv))
+        val fingerprintManager = FingerprintManagerCompat.from(context)
+        val signal = CancellationSignal()
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:79]: I would inline this as it is only used in `onAuthenticationError` to keep the initialisation of stuff as minimal as necessary.

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt [hunk 12]
```diff
@@ -68,53 +71,35 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
...
+        return suspendCoroutine { cont ->
+            val callback = object:  FingerprintManagerCompat.AuthenticationCallback() {
+                override fun onAuthenticationSucceeded(result: FingerprintManagerCompat.AuthenticationResult?) {
+                    result?.cryptoObject?.cipher?.let {
+                        cont.resume(AuthenticationResultSuccess(it))
+                    } ?: run {
+                        cont.resumeWithException(IllegalStateException("Ciper is null"))
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:86]: Typo on `Cipher` although I would spell it `cypher`. Both seem to be correct :D

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt [hunk 13]
```diff
@@ -68,53 +71,35 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
...
-                    if (!signal.isCanceled) {
-                        signal.cancel()
-                        this.emitter = null
+    @SuppressLint("MissingPermission")
+    @RequiresApi(Build.VERSION_CODES.M)
+    override suspend fun authenticate(iv: ByteArray?): AuthenticationResult {
+        val cryptoObject = FingerprintManagerCompat.CryptoObject(createCipher(iv))
+        val fingerprintManager = FingerprintManagerCompat.from(context)
+        val signal = CancellationSignal()
+        return suspendCoroutine { cont ->
```
Comment (jpalvarezl) [inline security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:80]: Here you are returning the `coroutine` but the returning type is `AuthenticationResult` ? I am a bit confused as to what is happening here.

File: security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt [hunk 14]
```diff
@@ -60,84 +63,65 @@ class AesEncryptionManagerTest {
...
+        // Check that device can be unlocked with old password
+        assertTrue(manager.unlockWithPassword(("test2".toByteArray())))
 
         // Check that data encrypted with old password can still be decrypted
         assertEquals("Hello World", String(manager.decrypt(encryptedData)))
     }
 
     @Test
-    fun fingerprintFlow() {
+    fun fingerprintFlow() = runBlocking {
```
Comment (jpalvarezl) [inline security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt:109]: Maybe we should be using `runBlockingTest`? I think the only difference is that it ensures the serialised execution of tests.

File: utils-testing-coroutines/src/main/AndroidManifest.xml [hunk 15]
```diff
@@ -0,0 +1 @@
+<manifest package="pm.gnosis.tests.utils.coroutines" />
```
Comment (jpalvarezl) [inline utils-testing-coroutines/src/main/AndroidManifest.xml:1]: Maybe we don't need a manifest and we could make this module `kotlin/Java`  only from the get-go?

File: utils-testing-rx/src/main/AndroidManifest.xml [hunk 16]
```diff
@@ -0,0 +1 @@
+<manifest package="pm.gnosis.tests.utils" />
```
Comment (jpalvarezl) [inline utils-testing-rx/src/main/AndroidManifest.xml:1]: Same here. This module doesn't necessarily have to be an android module. If we keep it as a `Java library` we can re-use it and we generate artefacts that are not android dependent.

**Review by jpalvarezl — COMMENTED** [review]
Looks nice and more readable. Except for a couple of things, that I am not sure are done like that by design, I would say is really close to being done!

File: security/src/main/java/pm/gnosis/svalinn/security/EncryptionManager.kt [hunk 17]
```diff
@@ -1,23 +1,20 @@
...
-    fun observeFingerprintForSetup(): Observable<Boolean>
-    fun observeFingerprintForUnlock(): Observable<FingerprintUnlockResult>
-    fun clearFingerprintData(): Completable
-    fun isFingerPrintSet(): Single<Boolean>
+    fun setupPassword(newPassword: ByteArray, oldPassword: ByteArray? = null): Boolean
+    fun initialized(): Boolean
+    suspend fun setupFingerprint(): Boolean
+    suspend fun unlockWithFingerprint(): FingerprintUnlockResult
+    fun clearFingerprintData()
+    fun isFingerPrintSet(): Boolean
```
[the remark this answers is not in the document]
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/EncryptionManager.kt:17]: The async call can be done on the class that calls this method. The fingerprint methods require handling of callbacks and therefore need to be suspend methods.

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt [hunk 18]
```diff
@@ -69,10 +65,8 @@ class AesEncryptionManager(
         return randomBytes
     }
 
-    override fun initialized(): Single<Boolean> {
-        return Single.fromCallable {
-            preferencesManager.prefs.getString(PREF_KEY_PASSWORD_ENCRYPTED_APP_KEY, null) != null
-        }
+    override fun initialized(): Boolean {
+        return preferencesManager.prefs.getString(PREF_KEY_PASSWORD_ENCRYPTED_APP_KEY, null) != null
```
[the remark this answers is not in the document]
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:69]: In general I would agree, but this PR was only about removing rx, not about refactoring it to work in a non android context ;)

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt (hunk 1 above)
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:79]: Imo that is up to the developer that is using the lib ... not sure if I would enforce this if not 100% necessary

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt (hunk 4 above)
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:202]: I just copy pasted code around without looking at the style

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt (hunk 7 above)
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:223]: We have a test for this: expected behaviour is that is throws

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt (hunk 10 above)
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:76]: iv is the crypto iv (part of the encrypted file that ensure that bruteforcing is harder)

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt (hunk 11 above)
Comment (rmeissner) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:79]: Yeah cancelation is not handled right now ... this is a todo

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt (hunk 1 above)
Comment (jpalvarezl) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:79]: I was thinking though, the dev could still execute this in the main thread (if for some reason necessary) but then it would also have to be explicit from which context you launch the coroutine. This is just a nudge, kind of like saying "maybe you should do this in a different thread". Either way, it is not super important.

File: android-common/src/main/java/pm/gnosis/svalinn/common/utils/QrCodeUtils.kt [hunk 19]
```diff
@@ -8,21 +8,12 @@ import androidx.fragment.app.Fragment
 import com.google.zxing.BarcodeFormat
 import com.google.zxing.WriterException
 import com.google.zxing.qrcode.QRCodeWriter
-import io.reactivex.Single
-import io.reactivex.schedulers.Schedulers
 
 interface QrCodeGenerator {
-    fun generateQrCode(contents: String, width: Int = 512, height: Int = 512, @ColorInt backgroundColor: Int = Color.WHITE): Single<Bitmap>
     fun generateQrCodeSync(contents: String, width: Int, height: Int, @ColorInt backgroundColor: Int): Bitmap
```
Comment (biafra23) [inline android-common/src/main/java/pm/gnosis/svalinn/common/utils/QrCodeUtils.kt:13]: If that is the only one, we can probably call it `generateQrCode()`

File: gradlew [hunk 20]
```diff
@@ -1,25 +1,43 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (biafra23) [inline gradlew:1]: Do we need to change this to a possibly incompatible shell? sh can be bash, but on some systems it might not be. Oh wait, is this due to gradle wrapper update?

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt (hunk 4 above)
Comment (biafra23) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AesEncryptionManager.kt:211]: I want to do this.

File: security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt (hunk 11 above)
Comment (biafra23) [reply security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt:79]: Is `authenticate()`one of the methods to disable? We're not using the FingerPrinHelper ATM. If not. I dont't understand what exactly is wrong with the cancellation.

File: security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt (hunk 14 above)
Comment (biafra23) [reply security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt:109]: `runBlockingTest` doesn't wait if there is a `delay`. So IMHO we should prefer it for tests.

File: utils-testing-coroutines/src/main/AndroidManifest.xml (hunk 15 above)
Comment (biafra23) [reply utils-testing-coroutines/src/main/AndroidManifest.xml:1]: Future PR?

Comment (jpalvarezl) [reply utils-testing-coroutines/src/main/AndroidManifest.xml:1]: Sure :) we can make this part of a bigger effort to decouple the android bits from the kotlin bits :)

File: security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt [hunk 21]
```diff
@@ -54,279 +47,56 @@ class AesEncryptionManagerTest {
...
-        unlockObserver.assertNoErrors().assertValue(true)
+        // Check that device can be unlocked with old password
+        assertTrue(manager.unlockWithPassword(("test2".toByteArray())))
 
         // Check that data encrypted with old password can still be decrypted
         assertEquals("Hello World", String(manager.decrypt(encryptedData)))
     }
 
-    @Test
-    fun fingerprintFlow() {
```
Comment (jpalvarezl) [inline security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt:124]: Did we get rid entirely of the fingerprint support? I am a bit confused. Will this be forked to a different issue?

File: utils-testing-coroutines/build.gradle [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+
+    //Kotlin
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:$versions.kotlin"
+
+    // Coroutines
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$versions.kotlinx_coroutines"
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:$versions.kotlinx_coroutines"
+
+    api "org.jetbrains.kotlinx:kotlinx-coroutines-test:$versions.kotlinx_coroutines"
+    api "androidx.arch.core:core-testing:2.1.0"
```
Comment (jpalvarezl) [inline utils-testing-coroutines/build.gradle:15]: should this not be `testImplementation`?

File: utils-testing-coroutines/build.gradle [hunk 23]
```diff
@@ -0,0 +1,16 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+dependencies {
+    implementation "junit:junit:$versions.junit"
```
Comment (jpalvarezl) [inline utils-testing-coroutines/build.gradle:5]: Should this not be `testImplementation` ? Or is it because this library is only meant to be consumed as a `testImplementation` dependency?

File: utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+
+    override fun starting(description: Description?) {
+        super.starting(description)
+        Dispatchers.setMain(TestCoroutineDispatcher())
+    }
+
+    override fun finished(description: Description?) {
+        Dispatchers.resetMain()
+        super.finished(description)
+    }
```
Comment (jpalvarezl) [inline utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt:22]: This file as well as `MainCoroutineScopeRule` are forcing the use of experimental API which I am not sure it is advisable for library dependency. Here I suppose we should aim rather for stability?

File: utils-testing-rx/build.gradle [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+    //Rx
+    implementation "io.reactivex.rxjava2:rxjava:$versions.rxjava"
+    implementation "io.reactivex.rxjava2:rxkotlin:$versions.rxkotlin"
+    implementation "io.reactivex.rxjava2:rxandroid:$versions.rxandroid"
+
+    // Mockito
+    implementation "org.mockito:mockito-core:$versions.mockito"
+    implementation "org.mockito:mockito-inline:$versions.mockito"
+
+    implementation "androidx.recyclerview:recyclerview:$versions.androidx_recycler_view"
```
Comment (jpalvarezl) [inline utils-testing-rx/build.gradle:19]: do we need `recycler_view` here O.O?

File: utils-testing-rx/build.gradle [hunk 26]
```diff
@@ -0,0 +1,20 @@
...
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:$versions.kotlin"
+
+    //Rx
+    implementation "io.reactivex.rxjava2:rxjava:$versions.rxjava"
+    implementation "io.reactivex.rxjava2:rxkotlin:$versions.rxkotlin"
+    implementation "io.reactivex.rxjava2:rxandroid:$versions.rxandroid"
+
+    // Mockito
+    implementation "org.mockito:mockito-core:$versions.mockito"
+    implementation "org.mockito:mockito-inline:$versions.mockito"
```
Comment (jpalvarezl) [inline utils-testing-rx/build.gradle:17]: No need for `testImplementation` instead? same goes for `junit`

File: utils-testing-coroutines/build.gradle (hunk 23 above)
Comment (biafra23) [reply utils-testing-coroutines/build.gradle:5]: This can be removed completely as there are no tests.

File: security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt (hunk 21 above)
Comment (biafra23) [reply security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt:124]: We will add Fingerprint support, when we need it.

File: utils-testing-rx/build.gradle (hunk 26 above)
Comment (biafra23) [reply utils-testing-rx/build.gradle:17]: Not needed at all.

Comment (biafra23) [reply utils-testing-rx/build.gradle:17]: junit is needed for building. There are no tests.

File: utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt (hunk 24 above)
Comment (biafra23) [reply utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt:22]: But this does not become part of any library that uses it? It is just used for testing, is it?

File: security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt (hunk 21 above)
Comment (rmeissner) [reply security/src/test/java/pm/gnosis/svalinn/security/impls/AesEncryptionManagerTest.kt:124]: maybe we should already create an issue for it :thinking:

File: utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt (hunk 24 above)
Comment (jpalvarezl) [reply utils-testing-coroutines/src/main/java/pm.gnosis.tests.utils/TestLifecycleRule.kt:22]: Well yeah that's true, this should be a `testImplementation` dependency somewhere. Now that you mention it, I guess this would have to be purposefully imported by someone anyhow so there is no need to worry about that.

File: utils-testing/src/main/java/pm.gnosis.tests.utils/Asserts.kt [hunk 27]
```diff
@@ -3,7 +3,7 @@ package pm.gnosis.tests.utils
 import org.junit.Assert
 
 object Asserts {
-    fun assertThrow(test: () -> Unit, message: String? = null, throwablePredicate: ((Throwable) -> Boolean)? = null) {
+    inline fun assertThrow(test: () -> Unit, message: String? = null, noinline throwablePredicate: ((Throwable) -> Boolean)? = null) {
```
Comment (jpalvarezl) [inline utils-testing/src/main/java/pm.gnosis.tests.utils/Asserts.kt:6]: Is this change necessary? I am not sure that there is a clear benefit of making this function `inline`

https://kotlinlang.org/docs/reference/inline-functions.html#noinline

**Review by jpalvarezl — APPROVED** [review]
Just one final comment on which I don't have a strong opinion! Good work 👍
